src/infra/core/tests/tests/test_compact_service_impl.rs

```diff
@@ -0,0 +1,778 @@
+                +--------+----+----------------------+----------------------+------+------------+
+                | 198    | 0  | 2050-01-01T12:00:00Z | 2020-04-09T00:00:00Z | A    | 1000       |
+                | 199    | 0  | 2050-01-01T12:00:00Z | 2020-04-10T00:00:00Z | B    | 2000       |
+                +--------+----+----------------------+----------------------+------+------------+
+                "#
+            ),
+        )
+        .await;
+
+    assert_matches!(
+        harness
+            .compact_svc
+            .compact_dataset(
+                &dataset_handle,
+                MAX_SLICE_SIZE,
+                10,
+                Some(Arc::new(NullCompactionMultiListener {}))
+            )
+            .await,
+        Ok(CompactResult::Finished),
+    );
+
+    let new_blocks = harness
+        .get_dataset_blocks(&root_dataset_alias.as_local_ref())
+        .await;
+
+    // We compacted all data slices and blocks into 20
+    assert_eq!(new_blocks.len(), 24);
+
+    // check the last block data and offsets
+    data_helper
+        .assert_last_data_eq(
+            indoc!(
+                r#"
+             message arrow_schema {
+               OPTIONAL INT64 offset;
+               REQUIRED INT32 op;
+               REQUIRED INT64 system_time (TIMESTAMP(MILLIS,true));
+               OPTIONAL INT64 date (TIMESTAMP(MILLIS,true));
+               OPTIONAL BYTE_ARRAY city (STRING);
+               OPTIONAL INT64 population;
+             }
+             "#
+            ),
+            indoc!(
+                r#"
+             +--------+----+----------------------+----------------------+------+------------+
+             | offset | op | system_time          | date                 | city | population |
+             +--------+----+----------------------+----------------------+------+------------+
+             | 190    | 0  | 2050-01-01T12:00:00Z | 2020-04-05T00:00:00Z | A    | 1000       |
+             | 191    | 0  | 2050-01-01T12:00:00Z | 2020-04-06T00:00:00Z | B    | 2000       |
+             | 192    | 0  | 2050-01-01T12:00:00Z | 2020-04-06T00:00:00Z | A    | 1000       |
+             | 193    | 0  | 2050-01-01T12:00:00Z | 2020-04-07T00:00:00Z | B    | 2000       |
+             | 194    | 0  | 2050-01-01T12:00:00Z | 2020-04-07T00:00:00Z | A    | 1000       |
+             | 195    | 0  | 2050-01-01T12:00:00Z | 2020-04-08T00:00:00Z | B    | 2000       |
+             | 196    | 0  | 2050-01-01T12:00:00Z | 2020-04-08T00:00:00Z | A    | 1000       |
+             | 197    | 0  | 2050-01-01T12:00:00Z | 2020-04-09T00:00:00Z | B    | 2000       |
+             | 198    | 0  | 2050-01-01T12:00:00Z | 2020-04-09T00:00:00Z | A    | 1000       |
+             | 199    | 0  | 2050-01-01T12:00:00Z | 2020-04-10T00:00:00Z | B    | 2000       |
+             +--------+----+----------------------+----------------------+------+------------+
+            "#
+            ),
+        )
+        .await;
```

Comment:
Can you please also add a call to `VerificationService` as the last step of this test to check consistency of the resulting dataset. I think over time we will add more validations to `VerificationService` such as checking that data offset ranges match the ones declared in metadata, that `op` sequences are valid and that events are sorted correctly... So this test will benefit from all those future checks.

Reply:
Added